### ISSUE TITLE
okta_factor resources error on `okta_email` factor email enrollment provider 

### ISSUE THREAD

**Issue by noinarisak** [okta/terraform-provider-okta#281]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```bash
Terraform v0.14.4
+ provider registry.terraform.io/oktadeveloper/okta v3.7.4

and, 

Terraform v0.13.6
+ provider registry.terraform.io/oktadeveloper/okta v3.7.4

```

### Affected Resource(s)

- okta_factor

### Terraform Configuration Files

```hcl
resource "okta_factor" "example" {
  provider_id = "okta_email"
}
```

### Debug Output

```bash
2021/01/12 12:07:37 [INFO] Terraform version: 0.14.4  
2021/01/12 12:07:37 [INFO] Go runtime version: go1.15.6
2021/01/12 12:07:37 [INFO] CLI args: []string{"/usr/local/Cellar/tfenv/1.0.2/versions/0.14.4/terraform", "plan"}
2021/01/12 12:07:37 [DEBUG] Attempting to open CLI config file: /Users/noinarisak/.terraformrc
2021/01/12 12:07:37 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2021/01/12 12:07:37 [DEBUG] checking for credentials in "/Users/noinarisak/.terraform.d/plugins"
2021/01/12 12:07:37 [DEBUG] checking for credentials in "/Users/noinarisak/.terraform.d/plugins/darwin_amd64"
2021/01/12 12:07:37 [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2021/01/12 12:07:37 [DEBUG] will search for provider plugins in /Users/noinarisak/.terraform.d/plugins
2021/01/12 12:07:37 [TRACE] getproviders.SearchLocalDirectory: /Users/noinarisak/.terraform.d/plugins is a symlink to /Users/noinarisak/.terraform.d/plugins
2021/01/12 12:07:37 [DEBUG] ignoring non-existing provider search directory /Users/noinarisak/Library/Application Support/io.terraform/plugins
2021/01/12 12:07:37 [DEBUG] ignoring non-existing provider search directory /Library/Application Support/io.terraform/plugins
2021/01/12 12:07:37 [INFO] CLI command args: []string{"plan"}
2021/01/12 12:07:37 [TRACE] Meta.Backend: no config given or present on disk, so returning nil config
2021/01/12 12:07:37 [TRACE] Meta.Backend: backend has not previously been initialized in this working directory
2021/01/12 12:07:37 [DEBUG] New state was assigned lineage "d957c267-0cce-ea6b-ef80-c892c582328b"
2021/01/12 12:07:37 [TRACE] Meta.Backend: using default local state only (no backend configuration, and no existing initialized backend)
2021/01/12 12:07:37 [TRACE] Meta.Backend: instantiated backend of type <nil>
2021/01/12 12:07:37 [TRACE] providercache.fillMetaCache: scanning directory .terraform/providers
2021/01/12 12:07:37 [TRACE] getproviders.SearchLocalDirectory: .terraform/providers is a symlink to .terraform/providers
2021/01/12 12:07:37 [TRACE] getproviders.SearchLocalDirectory: found registry.terraform.io/oktadeveloper/okta v3.7.4 for darwin_amd64 at .terraform/providers/registry.terraform.io/oktadeveloper/okta/3.7.4/darwin_amd64
2021/01/12 12:07:37 [TRACE] providercache.fillMetaCache: including .terraform/providers/registry.terraform.io/oktadeveloper/okta/3.7.4/darwin_amd64 as a candidate package for registry.terraform.io/oktadeveloper/okta 3.7.4
2021/01/12 12:07:37 [DEBUG] checking for provisioner in "."
2021/01/12 12:07:37 [DEBUG] checking for provisioner in "/usr/local/Cellar/tfenv/1.0.2/versions/0.14.4"
2021/01/12 12:07:37 [DEBUG] checking for provisioner in ".terraform/plugins/darwin_amd64"
2021/01/12 12:07:37 [DEBUG] checking for provisioner in "/Users/noinarisak/.terraform.d/plugins"
2021/01/12 12:07:37 [DEBUG] checking for provisioner in "/Users/noinarisak/.terraform.d/plugins/darwin_amd64"
2021/01/12 12:07:37 [TRACE] Meta.Backend: backend <nil> does not support operations, so wrapping it in a local backend
2021/01/12 12:07:37 [INFO] backend/local: starting Plan operation
2021/01/12 12:07:37 [TRACE] backend/local: requesting state manager for workspace "default"
2021/01/12 12:07:37 [TRACE] backend/local: state manager for workspace "default" will:
 - read initial snapshot from terraform.tfstate
 - write new snapshots to terraform.tfstate
 - create any backup at terraform.tfstate.backup
2021/01/12 12:07:37 [TRACE] backend/local: requesting state lock for workspace "default"
2021/01/12 12:07:37 [TRACE] statemgr.Filesystem: preparing to manage state snapshots at terraform.tfstate
2021/01/12 12:07:37 [TRACE] statemgr.Filesystem: no previously-stored snapshot exists
2021/01/12 12:07:37 [TRACE] statemgr.Filesystem: locking terraform.tfstate using fcntl flock
2021/01/12 12:07:37 [TRACE] statemgr.Filesystem: writing lock metadata to .terraform.tfstate.lock.info
2021/01/12 12:07:37 [TRACE] backend/local: reading remote state for workspace "default"
2021/01/12 12:07:37 [TRACE] statemgr.Filesystem: reading latest snapshot from terraform.tfstate
2021/01/12 12:07:37 [TRACE] statemgr.Filesystem: snapshot file has nil snapshot, but that's okay
2021/01/12 12:07:37 [TRACE] statemgr.Filesystem: read nil snapshot
2021/01/12 12:07:37 [TRACE] backend/local: retrieving local state snapshot for workspace "default"
2021/01/12 12:07:37 [TRACE] backend/local: building context for current working directory
2021/01/12 12:07:37 [TRACE] terraform.NewContext: starting
2021/01/12 12:07:37 [TRACE] terraform.NewContext: loading provider schemas
2021/01/12 12:07:37 [TRACE] LoadSchemas: retrieving schema for provider type "registry.terraform.io/oktadeveloper/okta"
2021-01-12T12:07:37.656-0600 [INFO]  plugin: configuring client automatic mTLS
2021-01-12T12:07:37.680-0600 [DEBUG] plugin: starting plugin: path=.terraform/providers/registry.terraform.io/oktadeveloper/okta/3.7.4/darwin_amd64/terraform-provider-okta_v3.7.4 args=[.terraform/providers/registry.terraform.io/oktadeveloper/okta/3.7.4/darwin_amd64/terraform-provider-okta_v3.7.4]
2021-01-12T12:07:37.685-0600 [DEBUG] plugin: plugin started: path=.terraform/providers/registry.terraform.io/oktadeveloper/okta/3.7.4/darwin_amd64/terraform-provider-okta_v3.7.4 pid=85495
2021-01-12T12:07:37.685-0600 [DEBUG] plugin: waiting for RPC address: path=.terraform/providers/registry.terraform.io/oktadeveloper/okta/3.7.4/darwin_amd64/terraform-provider-okta_v3.7.4
2021-01-12T12:07:37.706-0600 [INFO]  plugin.terraform-provider-okta_v3.7.4: configuring server automatic mTLS: timestamp=2021-01-12T12:07:37.705-0600
2021-01-12T12:07:37.730-0600 [DEBUG] plugin.terraform-provider-okta_v3.7.4: plugin address: address=/var/folders/mc/2zlfw6px5n167sdm76v5qngh0000gn/T/plugin598739266 network=unix timestamp=2021-01-12T12:07:37.730-0600
2021-01-12T12:07:37.731-0600 [DEBUG] plugin: using plugin: version=5
2021/01/12 12:07:37 [TRACE] GRPCProvider: GetSchema
2021-01-12T12:07:37.784-0600 [TRACE] plugin.stdio: waiting for stdio data
2021/01/12 12:07:37 [TRACE] GRPCProvider: Close
2021-01-12T12:07:37.795-0600 [WARN]  plugin.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2021-01-12T12:07:37.797-0600 [DEBUG] plugin: plugin process exited: path=.terraform/providers/registry.terraform.io/oktadeveloper/okta/3.7.4/darwin_amd64/terraform-provider-okta_v3.7.4 pid=85495
2021-01-12T12:07:37.797-0600 [DEBUG] plugin: plugin exited
2021/01/12 12:07:37 [TRACE] terraform.NewContext: complete
2021/01/12 12:07:37 [TRACE] backend/local: finished building terraform.Context
2021/01/12 12:07:37 [TRACE] backend/local: requesting interactive input, if necessary
2021/01/12 12:07:37 [TRACE] Context.Input: Prompting for provider arguments
2021/01/12 12:07:37 [TRACE] Context.Input: Provider provider.okta declared at okta.tf:11,1-16
2021/01/12 12:07:37 [TRACE] Context.Input: Input for provider.okta: map[string]cty.Value{}
2021/01/12 12:07:37 [TRACE] backend/local: running validation operation
2021/01/12 12:07:37 [INFO] terraform: building graph: GraphTypeValidate
2021/01/12 12:07:37 [TRACE] Executing graph transform *terraform.ConfigTransformer
2021/01/12 12:07:37 [TRACE] ConfigTransformer: Starting for path: 
2021/01/12 12:07:37 [TRACE] Completed graph transform *terraform.ConfigTransformer with new graph:
  okta_factor.example - *terraform.NodeValidatableResource
  ------
2021/01/12 12:07:37 [TRACE] Executing graph transform *terraform.RootVariableTransformer
2021/01/12 12:07:37 [TRACE] Completed graph transform *terraform.RootVariableTransformer (no changes)
2021/01/12 12:07:37 [TRACE] Executing graph transform *terraform.ModuleVariableTransformer
2021/01/12 12:07:37 [TRACE] Completed graph transform *terraform.ModuleVariableTransformer (no changes)
2021/01/12 12:07:37 [TRACE] Executing graph transform *terraform.LocalTransformer
2021/01/12 12:07:37 [TRACE] Completed graph transform *terraform.LocalTransformer (no changes)
2021/01/12 12:07:37 [TRACE] Executing graph transform *terraform.OutputTransformer
2021/01/12 12:07:37 [TRACE] Completed graph transform *terraform.OutputTransformer (no changes)
2021/01/12 12:07:37 [TRACE] Executing graph transform *terraform.OrphanResourceInstanceTransformer
2021/01/12 12:07:37 [TRACE] Completed graph transform *terraform.OrphanResourceInstanceTransformer (no changes)
2021/01/12 12:07:37 [TRACE] Executing graph transform *terraform.StateTransformer
2021/01/12 12:07:37 [TRACE] StateTransformer: state is empty, so nothing to do
2021/01/12 12:07:37 [TRACE] Completed graph transform *terraform.StateTransformer (no changes)
2021/01/12 12:07:37 [TRACE] Executing graph transform *terraform.AttachStateTransformer
2021/01/12 12:07:37 [TRACE] Completed graph transform *terraform.AttachStateTransformer (no changes)
2021/01/12 12:07:37 [TRACE] Executing graph transform *terraform.OrphanOutputTransformer
2021/01/12 12:07:37 [TRACE] Completed graph transform *terraform.OrphanOutputTransformer (no changes)
2021/01/12 12:07:37 [TRACE] Executing graph transform *terraform.AttachResourceConfigTransformer
2021/01/12 12:07:37 [TRACE] AttachResourceConfigTransformer: attaching to "okta_factor.example" (*terraform.NodeValidatableResource) config from okta.tf:29,1-33
2021/01/12 12:07:37 [TRACE] AttachResourceConfigTransformer: attaching provider meta configs to okta_factor.example
2021/01/12 12:07:37 [TRACE] Completed graph transform *terraform.AttachResourceConfigTransformer (no changes)
2021/01/12 12:07:37 [TRACE] Executing graph transform *terraform.MissingProvisionerTransformer
2021/01/12 12:07:37 [TRACE] Completed graph transform *terraform.MissingProvisionerTransformer (no changes)
2021/01/12 12:07:37 [TRACE] Executing graph transform *terraform.ProvisionerTransformer
2021/01/12 12:07:37 [TRACE] Completed graph transform *terraform.ProvisionerTransformer (no changes)
2021/01/12 12:07:37 [TRACE] Executing graph transform *terraform.graphTransformerMulti
2021/01/12 12:07:37 [TRACE] (graphTransformerMulti) Executing graph transform *terraform.ProviderConfigTransformer
2021/01/12 12:07:37 [TRACE] ProviderConfigTransformer: attaching to "provider[\"registry.terraform.io/oktadeveloper/okta\"]" provider configuration from okta.tf:11,1-16
2021/01/12 12:07:37 [TRACE] (graphTransformerMulti) Completed graph transform *terraform.ProviderConfigTransformer with new graph:
  okta_factor.example - *terraform.NodeValidatableResource
  provider["registry.terraform.io/oktadeveloper/okta"] - *terraform.NodeApplyableProvider
  ------
2021/01/12 12:07:37 [TRACE] (graphTransformerMulti) Executing graph transform *terraform.MissingProviderTransformer
2021/01/12 12:07:37 [TRACE] (graphTransformerMulti) Completed graph transform *terraform.MissingProviderTransformer (no changes)
2021/01/12 12:07:37 [TRACE] (graphTransformerMulti) Executing graph transform *terraform.ProviderTransformer
2021/01/12 12:07:37 [TRACE] ProviderTransformer: exact match for provider["registry.terraform.io/oktadeveloper/okta"] serving okta_factor.example
2021/01/12 12:07:37 [DEBUG] ProviderTransformer: "okta_factor.example" (*terraform.NodeValidatableResource) needs provider["registry.terraform.io/oktadeveloper/okta"]
2021/01/12 12:07:37 [TRACE] (graphTransformerMulti) Completed graph transform *terraform.ProviderTransformer with new graph:
  okta_factor.example - *terraform.NodeValidatableResource
    provider["registry.terraform.io/oktadeveloper/okta"] - *terraform.NodeApplyableProvider
  provider["registry.terraform.io/oktadeveloper/okta"] - *terraform.NodeApplyableProvider
  ------
2021/01/12 12:07:37 [TRACE] (graphTransformerMulti) Executing graph transform *terraform.PruneProviderTransformer
2021/01/12 12:07:37 [TRACE] (graphTransformerMulti) Completed graph transform *terraform.PruneProviderTransformer (no changes)
2021/01/12 12:07:37 [TRACE] (graphTransformerMulti) Executing graph transform *terraform.ParentProviderTransformer
2021/01/12 12:07:37 [TRACE] (graphTransformerMulti) Completed graph transform *terraform.ParentProviderTransformer (no changes)
2021/01/12 12:07:37 [TRACE] Completed graph transform *terraform.graphTransformerMulti with new graph:
  okta_factor.example - *terraform.NodeValidatableResource
    provider["registry.terraform.io/oktadeveloper/okta"] - *terraform.NodeApplyableProvider
  provider["registry.terraform.io/oktadeveloper/okta"] - *terraform.NodeApplyableProvider
  ------
2021/01/12 12:07:37 [TRACE] Executing graph transform *terraform.RemovedModuleTransformer
2021/01/12 12:07:37 [TRACE] Completed graph transform *terraform.RemovedModuleTransformer (no changes)
2021/01/12 12:07:37 [TRACE] Executing graph transform *terraform.AttachSchemaTransformer
2021/01/12 12:07:37 [TRACE] AttachSchemaTransformer: attaching resource schema to okta_factor.example
2021/01/12 12:07:37 [TRACE] AttachSchemaTransformer: attaching provider config schema to provider["registry.terraform.io/oktadeveloper/okta"]
2021/01/12 12:07:37 [TRACE] Completed graph transform *terraform.AttachSchemaTransformer (no changes)
2021/01/12 12:07:37 [TRACE] Executing graph transform *terraform.ModuleExpansionTransformer
2021/01/12 12:07:37 [TRACE] Completed graph transform *terraform.ModuleExpansionTransformer (no changes)
2021/01/12 12:07:37 [TRACE] Executing graph transform *terraform.ReferenceTransformer
2021/01/12 12:07:37 [DEBUG] ReferenceTransformer: "okta_factor.example" references: []
2021/01/12 12:07:37 [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/oktadeveloper/okta\"]" references: []
2021/01/12 12:07:37 [TRACE] Completed graph transform *terraform.ReferenceTransformer (no changes)
2021/01/12 12:07:37 [TRACE] Executing graph transform *terraform.AttachDependenciesTransformer
2021/01/12 12:07:37 [TRACE] Completed graph transform *terraform.AttachDependenciesTransformer (no changes)
2021/01/12 12:07:37 [TRACE] Executing graph transform *terraform.attachDataResourceDependenciesTransformer
2021/01/12 12:07:37 [TRACE] Completed graph transform *terraform.attachDataResourceDependenciesTransformer (no changes)
2021/01/12 12:07:37 [TRACE] Executing graph transform *terraform.TargetsTransformer
2021/01/12 12:07:37 [TRACE] Completed graph transform *terraform.TargetsTransformer (no changes)
2021/01/12 12:07:37 [TRACE] Executing graph transform *terraform.ForcedCBDTransformer
2021/01/12 12:07:37 [TRACE] Completed graph transform *terraform.ForcedCBDTransformer (no changes)
2021/01/12 12:07:37 [TRACE] Executing graph transform *terraform.CountBoundaryTransformer
2021/01/12 12:07:37 [TRACE] Completed graph transform *terraform.CountBoundaryTransformer with new graph:
  meta.count-boundary (EachMode fixup) - *terraform.NodeCountBoundary
    okta_factor.example - *terraform.NodeValidatableResource
    provider["registry.terraform.io/oktadeveloper/okta"] - *terraform.NodeApplyableProvider
  okta_factor.example - *terraform.NodeValidatableResource
    provider["registry.terraform.io/oktadeveloper/okta"] - *terraform.NodeApplyableProvider
  provider["registry.terraform.io/oktadeveloper/okta"] - *terraform.NodeApplyableProvider
  ------
2021/01/12 12:07:37 [TRACE] Executing graph transform *terraform.CloseProviderTransformer
2021/01/12 12:07:37 [TRACE] Completed graph transform *terraform.CloseProviderTransformer with new graph:
  meta.count-boundary (EachMode fixup) - *terraform.NodeCountBoundary
    okta_factor.example - *terraform.NodeValidatableResource
    provider["registry.terraform.io/oktadeveloper/okta"] - *terraform.NodeApplyableProvider
  okta_factor.example - *terraform.NodeValidatableResource
    provider["registry.terraform.io/oktadeveloper/okta"] - *terraform.NodeApplyableProvider
  provider["registry.terraform.io/oktadeveloper/okta"] - *terraform.NodeApplyableProvider
  provider["registry.terraform.io/oktadeveloper/okta"] (close) - *terraform.graphNodeCloseProvider
    okta_factor.example - *terraform.NodeValidatableResource
    provider["registry.terraform.io/oktadeveloper/okta"] - *terraform.NodeApplyableProvider
  ------
2021/01/12 12:07:37 [TRACE] Executing graph transform *terraform.CloseRootModuleTransformer
2021/01/12 12:07:37 [TRACE] Completed graph transform *terraform.CloseRootModuleTransformer with new graph:
  meta.count-boundary (EachMode fixup) - *terraform.NodeCountBoundary
    okta_factor.example - *terraform.NodeValidatableResource
    provider["registry.terraform.io/oktadeveloper/okta"] - *terraform.NodeApplyableProvider
  okta_factor.example - *terraform.NodeValidatableResource
    provider["registry.terraform.io/oktadeveloper/okta"] - *terraform.NodeApplyableProvider
  provider["registry.terraform.io/oktadeveloper/okta"] - *terraform.NodeApplyableProvider
  provider["registry.terraform.io/oktadeveloper/okta"] (close) - *terraform.graphNodeCloseProvider
    okta_factor.example - *terraform.NodeValidatableResource
    provider["registry.terraform.io/oktadeveloper/okta"] - *terraform.NodeApplyableProvider
  root - *terraform.nodeCloseModule
    meta.count-boundary (EachMode fixup) - *terraform.NodeCountBoundary
    provider["registry.terraform.io/oktadeveloper/okta"] (close) - *terraform.graphNodeCloseProvider
  ------
2021/01/12 12:07:37 [TRACE] Executing graph transform *terraform.TransitiveReductionTransformer
2021/01/12 12:07:37 [TRACE] Completed graph transform *terraform.TransitiveReductionTransformer with new graph:
  meta.count-boundary (EachMode fixup) - *terraform.NodeCountBoundary
    okta_factor.example - *terraform.NodeValidatableResource
  okta_factor.example - *terraform.NodeValidatableResource
    provider["registry.terraform.io/oktadeveloper/okta"] - *terraform.NodeApplyableProvider
  provider["registry.terraform.io/oktadeveloper/okta"] - *terraform.NodeApplyableProvider
  provider["registry.terraform.io/oktadeveloper/okta"] (close) - *terraform.graphNodeCloseProvider
    okta_factor.example - *terraform.NodeValidatableResource
  root - *terraform.nodeCloseModule
    meta.count-boundary (EachMode fixup) - *terraform.NodeCountBoundary
    provider["registry.terraform.io/oktadeveloper/okta"] (close) - *terraform.graphNodeCloseProvider
  ------
2021/01/12 12:07:37 [DEBUG] Starting graph walk: walkValidate
2021/01/12 12:07:37 [TRACE] dag/walk: visiting "provider[\"registry.terraform.io/oktadeveloper/okta\"]"
2021/01/12 12:07:37 [TRACE] vertex "provider[\"registry.terraform.io/oktadeveloper/okta\"]": starting visit (*terraform.NodeApplyableProvider)
2021-01-12T12:07:37.799-0600 [INFO]  plugin: configuring client automatic mTLS
2021-01-12T12:07:37.824-0600 [DEBUG] plugin: starting plugin: path=.terraform/providers/registry.terraform.io/oktadeveloper/okta/3.7.4/darwin_amd64/terraform-provider-okta_v3.7.4 args=[.terraform/providers/registry.terraform.io/oktadeveloper/okta/3.7.4/darwin_amd64/terraform-provider-okta_v3.7.4]
2021-01-12T12:07:37.828-0600 [DEBUG] plugin: plugin started: path=.terraform/providers/registry.terraform.io/oktadeveloper/okta/3.7.4/darwin_amd64/terraform-provider-okta_v3.7.4 pid=85496
2021-01-12T12:07:37.828-0600 [DEBUG] plugin: waiting for RPC address: path=.terraform/providers/registry.terraform.io/oktadeveloper/okta/3.7.4/darwin_amd64/terraform-provider-okta_v3.7.4
2021-01-12T12:07:37.848-0600 [INFO]  plugin.terraform-provider-okta_v3.7.4: configuring server automatic mTLS: timestamp=2021-01-12T12:07:37.848-0600
2021-01-12T12:07:37.871-0600 [DEBUG] plugin.terraform-provider-okta_v3.7.4: plugin address: address=/var/folders/mc/2zlfw6px5n167sdm76v5qngh0000gn/T/plugin255120559 network=unix timestamp=2021-01-12T12:07:37.871-0600
2021-01-12T12:07:37.871-0600 [DEBUG] plugin: using plugin: version=5
2021-01-12T12:07:37.917-0600 [TRACE] plugin.stdio: waiting for stdio data
2021/01/12 12:07:37 [TRACE] BuiltinEvalContext: Initialized "provider[\"registry.terraform.io/oktadeveloper/okta\"]" provider for provider["registry.terraform.io/oktadeveloper/okta"]
2021/01/12 12:07:37 [TRACE] buildProviderConfig for provider["registry.terraform.io/oktadeveloper/okta"]: using explicit config only
2021/01/12 12:07:37 [TRACE] GRPCProvider: GetSchema
2021/01/12 12:07:37 [TRACE] GRPCProvider: PrepareProviderConfig
2021/01/12 12:07:37 [TRACE] vertex "provider[\"registry.terraform.io/oktadeveloper/okta\"]": visit complete
2021/01/12 12:07:37 [TRACE] dag/walk: visiting "okta_factor.example"
2021/01/12 12:07:37 [TRACE] vertex "okta_factor.example": starting visit (*terraform.NodeValidatableResource)
2021/01/12 12:07:37 [TRACE] GRPCProvider: ValidateResourceTypeConfig
2021/01/12 12:07:37 [TRACE] vertex "okta_factor.example": visit complete
2021/01/12 12:07:37 [TRACE] dag/walk: visiting "meta.count-boundary (EachMode fixup)"
2021/01/12 12:07:37 [TRACE] vertex "meta.count-boundary (EachMode fixup)": starting visit (*terraform.NodeCountBoundary)
2021/01/12 12:07:37 [TRACE] vertex "meta.count-boundary (EachMode fixup)": visit complete
2021/01/12 12:07:37 [TRACE] dag/walk: visiting "provider[\"registry.terraform.io/oktadeveloper/okta\"] (close)"
2021/01/12 12:07:37 [TRACE] vertex "provider[\"registry.terraform.io/oktadeveloper/okta\"] (close)": starting visit (*terraform.graphNodeCloseProvider)
2021/01/12 12:07:37 [TRACE] GRPCProvider: Close
2021-01-12T12:07:37.935-0600 [WARN]  plugin.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2021-01-12T12:07:37.937-0600 [DEBUG] plugin: plugin process exited: path=.terraform/providers/registry.terraform.io/oktadeveloper/okta/3.7.4/darwin_amd64/terraform-provider-okta_v3.7.4 pid=85496

2021-01-12T12:07:37.938-0600 [DEBUG] plugin: plugin exited
Error: expected [{{} provider_id}] to be one of duo,fido_u2f,fido_webauthn,google_otp,okta_call,okta_otp,okta_push,okta_question,okta_sms,rsa_token,symantec_vip,yubikey_token, got okta_email

2021/01/12 12:07:37 [TRACE] vertex "provider[\"registry.terraform.io/oktadeveloper/okta\"] (close)": visit complete
2021/01/12 12:07:37 [TRACE] dag/walk: visiting "root"
2021/01/12 12:07:37 [TRACE] vertex "root": starting visit (*terraform.nodeCloseModule)
2021/01/12 12:07:37 [TRACE] vertex "root": visit complete
2021/01/12 12:07:37 [TRACE] statemgr.Filesystem: removing lock metadata file .terraform.tfstate.lock.info
2021/01/12 12:07:37 [TRACE] statemgr.Filesystem: unlocking terraform.tfstate using fcntl flock
  on okta.tf line 30, in resource "okta_factor" "example":
  30:   provider_id = "okta_email"

```
### Expected Behavior

Create email factor resource. ie Activated email factor enrollment.

### Actual Behavior

```bash
Error: expected [{{} provider_id}] to be one of duo,fido_u2f,fido_webauthn,google_otp,okta_call,okta_otp,okta_push,okta_question,okta_sms,rsa_token,symantec_vip,yubikey_token, got okta_email

  on okta.tf line 30, in resource "okta_factor" "example":
  30:   provider_id = "okta_email"
```

### Steps to Reproduce

1. `terraform init`
2. `terraform plan`

### References

- #269

**Comment by noinarisak** [comment]
We should also update the associated [doc](https://registry.terraform.io/providers/oktadeveloper/okta/latest/docs/resources/factor), since `okta_email` is missing in the `provider_id` list.

**Comment by bogdanprodan-okta** [comment]
This bug will be fixed in the upcoming 3.8.0 version
